client/auth/login.jsx

```diff
@@ -167,14 +181,32 @@ module.exports = React.createClass( {
 					</FormFieldset>
 					<FormButtonsBar>
 						<FormButton disabled={ ! this.canSubmitForm() } onClick={ this.recordClickEvent( 'Sign in' ) } >
-							{ requires2fa ? this.translate( 'Verify' ) : this.translate( 'Sign in' ) }
+							{ requires2fa === 'code' ? this.translate( 'Verify' ) : this.translate( 'Sign in' ) }
 						</FormButton>
 					</FormButtonsBar>
 					{ ! requires2fa && <LostPassword /> }
-					{ errorMessage && <Notice text={ errorMessage } status={ errorLevel } showDismiss={ false } /> }
-					{ requires2fa && <AuthCodeButton username={ this.state.login } password={ this.state.password } /> }
+					{ errorMessage &&
+						<Notice text={ errorMessage } status={ errorLevel } showDismiss={ false } />
+					}
+					{ ( requires2fa === 'push-verification' ) &&
+						<Interval onTick={ this.verifyPushAuthentication } period={ EVERY_FIVE_SECONDS }>
+							<FormButtonsBar>
+								<FormButton ref="useAuthCode" onClick={ this.useAuthCode } >
+									{ this.translate( 'Verify with code instead' ) }
+								</FormButton>
+							</FormButtonsBar>
+						</Interval>
```

Comment:
Yes, I am quite happy with how much boilerplate it eliminated. I liked that it groups the callback execution to reduce device battery consumption, remarkable! 👍